Allow players to choose marker, stop using atoms

ksabbak opened the pull request:
Players can now choose their markers and the marker of the computer opponent. The marker must be one character in length and cannot be the same as the other marker.

Player status is now being determined by a "turn" instead of clojure's atom data structure for a more functional approach.

ksabbak committed:
Allow players to choose marker, stop using atoms

Players can now choose their markers and the marker of the computer opponent. The marker must be one character in length and cannot be the same as the other marker.

Player status is now being determined by a "turn" instead of clojure's atom data structure for a more functional approach.

hnlee commented on src/clojure_tictactoe/game/game_play.clj:
```diff
@@ -8,27 +8,28 @@
+  (board/mark-space (move-function board) marker board))
+
+(defn continue-game [board markers turn]
+  (let [new-board (player-move (players/choose-player-function turn) board (markers (players/current-player turn)))]
```
Newlines here would make this easier to follow.

BWoodfork replied:
+1

ksabbak replied:
👍

hnlee commented on src/clojure_tictactoe/game/game_play.clj:
```diff
@@ -8,27 +8,28 @@
   (input-getter/continue-to-game)
-  (let [board (board/render-empty-board 9)]
+  (let [board (board/render-empty-board 9)
+        markers (players/acquire-both-markers '("you" "the computer"))]
```
Consider storing these hard coded values in a constant.

ksabbak replied:
👍

hnlee commented on src/clojure_tictactoe/game/game_play.clj:
```diff
@@ -8,27 +8,28 @@
+      (recur new-board markers (+ 1 turn)))
+    (if-let [results (rules/assess-winner board)]
+      (println (end-printer/game-won-message results))
+      (println end-printer/game-tie-message))))
```
Re: the use of `println`, should `game-play` know that the UI is a console?

ksabbak replied:
I agree it shouldn't, but the only solution I've been able to come up with so far is basically coming up with a wrapper that lives in one of the printers and just calls `println` and I don't love that idea either. Do you have any suggestions?

hnlee replied:
A simple wrapper around `println` doesn't sound too bad to me, honestly. It might seem excessive right now, but it would make it really easy to swap out for a different output function.

ksabbak replied:
Cool, I'll work this in with the next stories.

hnlee reviewed (changes requested):
Overall, continuing to look better and better.

Food for thought: what happens if the user inputs a character like "1" for a player marker? How should that be handled?

BWoodfork commented on src/clojure_tictactoe/game/players.clj:
```diff
@@ -2,20 +2,34 @@
+(defn current-player [turn]
+  (if (even? turn)
+    0
+    1))
```
What do these magic numbers mean?

ksabbak replied:
Fixed.

ksabbak committed:
Fix formatting, styles, etc.

Mostly in response to comments on PR 15
https://github.com/ksabbak/ClojureTTT/pull/15

Some in response to comments in PR 14
https://github.com/ksabbak/ClojureTTT/pull/14
